lib/Basics/FixedSizeAllocator.h

```diff
@@ -125,14 +119,20 @@ class FixedSizeAllocator {
     }
   }
 
-  void clear() {
-    _head.reset();
+  void clear() noexcept {
+    auto* block = _head;
+    while (block != nullptr) {
+      auto* next = block->getNextBlock();
+      block->~MemoryBlock();
+      ::operator delete(block);
+      block = next;
+    }
```

Comment:
I would prefer if this method would also set `_head` to a nullptr after having performed the cleanup. That will leave things in a consistent state, even though right now `clear()` is only called from the destructor.

Reply:
Good point!